### DISCUSSION TITLE
MEMCM / SCCM package generator

### DISCUSSION THREAD

**Discussion by noyard** [microsoft/winget-cli#1211] [General]
I have created a PowerShell script to create MEMCM Applications for a test lab using Winget package repository as a source.  My only issue is the install command.

How can I get the install commands from a Winget package?

Example is 
winget show "adobe acrobat reader dc"  
Shows the download URL and information needed to create the MEMCM Application.  but does not have the installer switches.

InstallerSwitches:
  Silent: /sAll /rs /rps /l /re
  SilentWithProgress: /sAll /rs /rps /l /re

**Comment by denelon** [comment]
We don't surface all of the manifest data using the "show" command. We're discussing when we will have a REST API based source with the contents of the community repository. Until then, the only other source is the GitHub repository https://aka.ms/winget-pkgs.

The show command makes some assumptions about which installer node to use, and that could impact switches it displayed based on the machine and user settings.

**Comment by endpointmanager** [comment]
@noyard: Please take a look on one of my projects... https://github.com/endpointmanager/wingetbridge-factory 
This could be for you if you are using MEMCM. It analyzes MSI, NULLSOFT, INNO (and very soon "BURN"-installers) from the winget repository "on the fly"... you just configure a list of PackageIds (e.g. "Microsoft.PowerBI") and they will be "synchronized" into MEMCM like magic.

**Reply by noyard** [reply]
I will take a look. I ended up using evergreen (https://stealthpuppy.com/evergreen/) to obtain the executables ... It even has software that Winget doesn't have...yet

**Reply by Masamune3210** [reply]
Sounds like a good way to fix those programs that winget doesn't have. As long as they have a silent switch in their installer, either have no redists or handles it itself, and isn't a portable exe/zip it can be added easily
